nc/include/component/vulkan/PointLight.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "component/Component.h"
+#include "math/Vector4.h"
+#include "directx/math/DirectXMath.h"
+
+namespace nc { class Transform; }
+
+namespace nc::vulkan
+{
+    struct Attenuation
+    {
+        float constant = 1.0f;
+        float linear = 0.7f;
+        float quadratic = 1.8f;
+    };
+
+    Attenuation GetAttenuationFromRange(float range);
+
+    struct PointLightInfo
+    {
+        Vector4 pos = Vector4::Zero();
+        Vector4 ambient = Vector4{0.2f, 0.2f, 0.2f, 1.0f};
+        Vector4 diffuseColor = Vector4{0.75f, 0.75f, 0.75f, 1.0f};
+        Vector4 specularColor = Vector4{0.75f, 0.75f, 0.75f, 1.0f};
+        float diffuseIntensity = 10.0f;
+        float attConst = 1.0f;
+        float attLin = 0.22f;
+        float attQuad = 0.20f;
+        bool isInitialized = 0;
```

Comment:
I haven't pieced together why exactly this is here, but I don't love the bool. Its costing 16 bytes with the padding and doesn't look like it is really needed. Maybe I haven't gone far enough to reveal its true purpose, but it feels icky vicky.